src/ledger/LedgerTxnTests.cpp

```diff
@@ -2018,3 +2020,163 @@ TEST_CASE("LedgerTxnEntry and LedgerTxnHeader move assignment", "[ledgerstate]")
+        return keys;
+    };
+
+    auto writeEntries =
+        [&getTimeScope](Application& app, size_t numSigners,
+                        std::vector<LedgerEntry> const& accounts) {
+            CLOG(WARNING, "Ledger") << "Creating accounts";
+            LedgerTxn ltx(app.getLedgerTxnRoot());
+            {
+                auto timer = getTimeScope(app, numSigners, "create");
+                for (auto const& le : accounts)
+                {
+                    ltx.create(le);
+                }
+            }
+
+            CLOG(WARNING, "Ledger") << "Writing accounts";
+            {
+                auto timer = getTimeScope(app, numSigners, "write");
+                ltx.commit();
+            }
+        };
+
+    auto readEntriesAndUpdateLastModified =
+        [&getTimeScope](Application& app, size_t numSigners,
+                        std::vector<LedgerKey> const& accounts) {
+            CLOG(WARNING, "Ledger") << "Reading accounts";
+            LedgerTxn ltx(app.getLedgerTxnRoot());
+            {
+                auto timer = getTimeScope(app, numSigners, "read");
+                for (auto const& key : accounts)
+                {
+                    ++ltx.load(key).current().lastModifiedLedgerSeq;
+                }
+            }
+
+            CLOG(WARNING, "Ledger")
+                << "Writing accounts with unchanged signers";
+            {
+                auto timer = getTimeScope(app, numSigners, "rewrite");
+                ltx.commit();
+            }
+        };
+
+    auto runTest = [&](Config::TestDbMode mode, size_t numAccounts,
+                       size_t numSigners) {
+        VirtualClock clock;
+        Config cfg(getTestConfig(0, mode));
+        cfg.ENTRY_CACHE_SIZE = 0;
+        cfg.BEST_OFFERS_CACHE_SIZE = 0;
+        Application::pointer app = createTestApplication(clock, cfg);
+        app->start();
+
+        CLOG(WARNING, "Ledger")
+            << "Generating " << numAccounts << " accounts with " << numSigners
+            << " signers each";
+        auto accounts = generateEntries(numAccounts, numSigners);
+        auto keys = generateKeys(accounts);
+
+        writeEntries(*app, numSigners, accounts);
+        readEntriesAndUpdateLastModified(*app, numSigners, keys);
+
+        CLOG(WARNING, "Ledger")
+            << "Done (" << getTimeSpent(*app, numSigners, "create") << ", "
+            << getTimeSpent(*app, numSigners, "write") << ", "
+            << getTimeSpent(*app, numSigners, "read") << ", "
+            << getTimeSpent(*app, numSigners, "rewrite") << ")";
+    };
+
+    SECTION("sqlite")
+    {
+        SECTION("0 signers")
+        {
+            runTest(Config::TESTDB_ON_DISK_SQLITE, 100000, 0);
+        }
+        SECTION("10 signers")
+        {
+            runTest(Config::TESTDB_ON_DISK_SQLITE, 100000, 10);
+        }
+        SECTION("20 signers")
+        {
+            runTest(Config::TESTDB_ON_DISK_SQLITE, 100000, 20);
+        }
+    }
+
+#ifdef USE_POSTGRES
```

Comment:
you don't have to copy the section, you can just do something like
```c++
    SECTION("sqlite")
    {
        runtest(Config::TESTDB_ON_DISK_SQLITE);
    }
#ifdef USE_POSTGRES
    SECTION("postgresql")
    {
        runtests(Config::TESTDB_POSTGRESQL);
    }
#endif
```

Reply:
Good idea, done.